### PR TITLE
Feature/new scan unifier

### PR DESCRIPTION
**Pull request by fmirus**
This pull request contains the following features:
- removes the old scan unifier
- adds the new scan unifier, including the feature of parsing the subscribed topics names from yaml-files
- adds yaml-files for each robot for the use of scan unifier
- updates the yaml-file for amcl to use the scan unifier (necessary for the use of three laser scanners for cob4)

tested in simulation. still needs tests on cob4, so please do not merge yet.

### CODE REVIEW

**Comment by floweisshardt** [comment]
@ipa-mig shouldn't we rename the package to have to same prefix as all others in cob_navigation?

File: ipa_navigation_scan_unifier/src/scan_unifier_node.cpp [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	// store initialization time of the node 
+	ros::Time start = ros::Time::now();
+
+	ros::Duration delay(my_scan_unifier_node.getStartDelay());
+
+	ros::Rate rate(my_scan_unifier_node.getLoopRate());
+
+	// start the calculation at the earliest after a given time interval to avoid tf-crash
+	ROS_DEBUG_STREAM("scan unifier: sleep for " << my_scan_unifier_node.getStartDelay() << " seconds before start of scan unification");
+	ros::Duration(my_scan_unifier_node.getStartDelay()).sleep();
```
Comment (floweisshardt) [inline ipa_navigation_scan_unifier/src/scan_unifier_node.cpp]: sleep is not very good. Can't we use a tf.wait_for_transform?

Comment (mgruhler) [inline ipa_navigation_scan_unifier/src/scan_unifier_node.cpp]: +1

**Comment by mgruhler** [comment]
A rename to cob_scan_unifier would be my favorite.
Alternatively: cob_navigation_scan_unifier/cob_nav_scan_unifier

**Comment by floweisshardt** [comment]
+1 for cob_scan_unifier

**Comment by NadiaHG** [comment]
Tested on cob4-1 :ok:

File: cob_scan_unifier/package.xml [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
-    This has shown to provide better results for SLAM than using two laser scanners over a topic relay.
-  </description>
+  <name>cob_scan_unifier</name>
+  <version>0.0.0</version>
+  <description>The cob_scan_unifier package holds code to unify two or more laser-scans to one unified scan-message</description>
 
   <maintainer email="frm@ipa.fraunhofer.de">Florian Mirus</maintainer>
+
+
+  <license>Fraunhofer IPA</license>
```
Comment (mgruhler) [inline cob_scan_unifier/package.xml]: we should adjust the license tag here and in the .cpp/.h file.
@ipa-fmw, @abubeck LGPL or BSD?

File: cob_navigation_config/cob3-1/scan_unifier_config.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#Scan_unifier_config-parameters
+
+input_scans: ["scan_front","scan_rear"]
+start_delay: 2.0
```
Comment (mgruhler) [inline cob_navigation_config/cob3-1/scan_unifier_config.yaml]: start_delay is not required anymore, correct?
If so, please remove.

File: cob_scan_unifier/src/scan_unifier_node.cpp [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+		
+		my_scan_unifier_node.checkUnifieCondition();
+		
+		ros::spinOnce();
+		rate.sleep();
+
+	}
+
+	return 0;
+}
```
Comment (mgruhler) [inline cob_scan_unifier/src/scan_unifier_node.cpp]: @ipa-frm: Could you also fix the indentation in the .h/.cpp files? I.e. either tabs or spaces. Currently, they are mixed.

File: cob_scan_unifier/package.xml (hunk 2 above)
Comment (floweisshardt) [inline cob_scan_unifier/package.xml]: should be LGPL

File: cob_scan_unifier/include/cob_scan_unifier/scan_unifier_node.h [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2011 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *   All rightes reserved. \n\n
+ *
+ *****************************************************************
```
Comment (mgruhler) [inline cob_scan_unifier/include/cob_scan_unifier/scan_unifier_node.h]: @ipa-frm : one last thing, please adjust the copyright here and in the cpp file. Template is in ipa320/setup/templates

**Comment by mgruhler** [comment]
visual review okay, functionality tested

**Pull request merged**